File: services/frontend/src/domains/person.ts
import type { DomainConfig, FieldConfig } from './base'

export interface PersonAttrs {
  domain: 'person'
  relationship?: string | null
  company?: string | null
  job_title?: string | null
  email?: string | null
  phone?: string | null
  birthday?: string | null
}

const FIELDS: FieldConfig[] = [
  { key: 'relationship', label: 'Relationship', kind: 'text' },
  { key: 'company', label: 'Company', kind: 'text' },
  { key: 'job_title', label: 'Job title', kind: 'text' },
  { key: 'email', label: 'Email', kind: 'text' },
  { key: 'phone', label: 'Phone', kind: 'text' },
  { key: 'birthday', label: 'Birthday', kind: 'date' },
]

export const personConfig: DomainConfig<PersonAttrs> = {
  label: 'Person',
  statuses: ['active', 'inactive'],
  logTypes: ['conversation', 'call', 'meeting', 'gift', 'milestone'],
  fields: FIELDS,
  defaultAttributes: () => ({ domain: 'person' }),
  namePlaceholder: 'Full name',
  locationPlaceholder: 'City, neighborhood, ...',
}

File: libs/shared/src/aria_shared/models/entities/person.py
from datetime import date
from typing import ClassVar, Literal

from .base import BaseAttributes


class PersonAttrs(BaseAttributes):
    DOMAIN: ClassVar[str] = "person"
    VALID_STATUSES: ClassVar[set[str]] = {"active", "inactive"}
    LOG_TYPES: ClassVar[set[str]] = {"conversation", "call", "meeting", "gift", "milestone"}

    domain: Literal["person"] = "person"
    relationship: str | None = None  # "friend", "family", "colleague", "neighbor", ... — free text
    company: str | None = None
    job_title: str | None = None
    email: str | None = None
    phone: str | None = None
    birthday: date | None = None
